exonum-java-binding-core/src/test/java/com/exonum/binding/storage/indices/CheckedMapProofMatcher.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2018 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.storage.indices;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.common.proofs.map.flat.CheckedMapProof;
+import com.exonum.binding.common.proofs.map.flat.ProofStatus;
+import javax.annotation.Nullable;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+import org.hamcrest.core.IsEqual;
+
+class CheckedMapProofMatcher extends TypeSafeMatcher<CheckedMapProof> {
+
+  private final HashCode key;
+  @Nullable
+  private final String expectedValue;
+
+  private final Matcher<byte[]> keyMatcher;
+  private final Matcher<byte[]> valueMatcher;
+
+  private CheckedMapProofMatcher(HashCode key, @Nullable String expectedValue) {
+    this.key = checkNotNull(key);
+    this.expectedValue = expectedValue;
+    keyMatcher = IsEqual.equalTo(key.asBytes());
+    valueMatcher =
+        expectedValue != null ? IsEqual.equalTo(expectedValue.getBytes()) : IsEqual.equalTo(null);
+  }
+
+  @Override
+  protected boolean matchesSafely(CheckedMapProof checkedMapProof) {
+    // In case of null expectedValue the absence of the key is checked
```

Comment:
Please add some intermediate variables here (status/missingKeys/entries) — long chains are hard to read.